### ISSUE TITLE
WSL2: server process not terminated on exit

### ISSUE THREAD

**Issue by radu-matei** [microsoft/vscode-remote-release#699]
```
Version: 1.36.0-insider (user setup)
Commit: 15b7c86d86319b187a5ef255c170184a8519c5af
Date: 2019-06-14T10:37:04.235Z
Electron: 4.2.4
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18917
```

In WSL2, the VS Code Server process is not terminated when the VS Code instance is closed, and it is not reused when a new window is opened (after all previous ones are closed):

After I started two instances of VS Code and close them both, I can still see all of the processes associated with both instances, and the resource consumption is noticeable (both CPU and memory), until I manually kill the processes:

![image](https://user-images.githubusercontent.com/13103165/59557174-13d7d600-8fdc-11e9-99cd-cb94a3dafbf3.png)

Notice the resource consumption with all VS Code windows closed:

![image](https://user-images.githubusercontent.com/13103165/59557184-539ebd80-8fdc-11e9-94e6-5132f153d05c.png)

![image](https://user-images.githubusercontent.com/13103165/59557186-61544300-8fdc-11e9-8aaf-4c28780ac6e9.png)

**Comment by AlexanderLieret** [comment]
The same issue also occurs with the SSH version.

**Comment by benc-uk** [comment]
Yes seeing the same, system will be cluttered with left over node processes.
I'm getting in the habit of running `pkill node` but really this needs to be fixed

**Comment by aeschli** [comment]
https://github.com/microsoft/vscode-remote-release/issues/203 will help here

**Comment by radu-matei** [comment]
Should we close this in favor of #203 then?

**Comment by aeschli** [comment]
Fixed by changes in #203
